### ISSUE TITLE
AIR-Node Group

### ISSUE THREAD

**Issue by biroMann** [benrugg/AI-Render#100]
Hello, and thank you for the great job. I will tag and share an animation as soon as I complete it.
...
I saw in previous versions that an AiR-Node Group was automatically generated in Compositor, for what I see it has been removed.
I think that it was useful because with it you could switch on/off the AiR effect just disconnecting it from the node tree or muting it(in this case also deselecting "Run Automatically on Render"), in this way it wasn't necessary a new workspace section(AI Render), it was enough the common Compositor section plus the AiR-Node Group. 
The second function that I found interesting(and that now I don't know how to make) about AiR-Node Group was that in the node tree I was able to add other effects AFTER AiR-Node Group, for instance to make some adjustments on colors because some SD checkpoints modify slightly them.

So, in my opinion, just adding AiR-Node Group in Compositor without new AI Render section, was working better, but maybe it's just my experience. 

Thank again for this amazing add-on.

**Comment by benrugg** [comment]
Yeah, I did like the node group for the second reason you mentioned - the ability to add effects or color modifications after Stable Diffusion. I ended up removing it when I added upscaling, because Blender doesn't have a way of allowing a larger image in the compositor than whatever was rendered. (So as far as I can tell, there's no way to keep the workflow of a node group now that upscaling is a feature).

**Comment by biroMann** [comment]
Ok, thank you very much for the answer.
Not to reopen the issue, I see it's closed, but just to understand if it possible to find a solution.
It could work if the upscaled rendered image is read through a RenderResultNode inside an AirNodeGroup. Maybe the Group can have different resolutions?

**Comment by benrugg** [comment]
You can read it back into an image node, but the final image will still be the original size (so an upscaled 2048x2048 for example, would still be rendered as 512x512).

If you want to do something similar to the workflow you had before, you can manually load in the upscaled image through an image node, add any nodes after that, and connect it to the output. And then you'll just need to render something quickly at the final output size, and the compositor will kick in.

**Comment by biroMann** [comment]
Ok, thank you, I will connect an ImageNode with AirResult to ViewerNode and then make adjustments. 

Consider that like in my case, if you are working not on a single image but on a clip with multiple scenes and sequences in which you must intervene often, you have to find and change the image in ViewerNode, every time you change frame, the previous workflow was much leaner, you just hit F12 and see result. In future advances, in the case should be possible, I think integrate the previous workflow it would be better for animators. Thank you again, this tool is amazing.

**Comment by biroMann** [comment]
But after that I have to render another time the sequence. Consider that like in my case, if you are working not on a single image but on a clip with multiple scenes and sequences in which you must intervene often, you have to find and change the image in ViewerNode, every time you change frame just to see the result on the particular bunch of frames and then rerender, in the previous workflow  you just hit F12 and see result, if it's ok you keyframe adjustments and lunch the final render. In future advances, in the case should be possible, I think integrate the previous workflow it would be much better for animators. Anyway maybe I find a work around.
Thank you again.

**Comment by benrugg** [comment]
Yes, that is a good point. The new way isn't a good workflow for animations, if you want to apply any compositing effects after Stable Diffusion. 

I will reopen this issue as a feature request, and maybe someone can provide insight on how this would be possible with an upscaled image. 

(One thing I will suggest as a possibility is that you could render out all the Stable Diffusion images, and then bring the entire image strip into Blender as a node, to do the compositing afterward. I _think_ this is doable).

**Comment by benrugg** [comment]
Thank you for writing this up. And for anyone coming across this in the future who may want to work on this feature, the summary of the core issue is that Blender's compositor only works with the image resolution that was rendered, so in this case it would be before it has been upscaled.
